pkg/router/template/template_helper.go

```diff
@@ -31,9 +32,33 @@ func isTrue(s string) bool {
 	return v
 }
 
+var compiledRegexp sync.Map
```

Comment:
I was wondering if we needed garbage collection for this map.  I believe we do not as every use of `firstMatch` specifies a pattern that is a hard-coded in the template itself (as opposed to specifying a pattern from an annotation, for example), so the size of the map is bounded by a function of the static set of patterns that are used in `haproxy-config.template`.